diff --git a/dist/esm/eventHandling.native.js b/dist/esm/eventHandling.native.js
index 4b989393..b2e0609c 100644
--- a/dist/esm/eventHandling.native.js
+++ b/dist/esm/eventHandling.native.js
@@ -35,6 +35,24 @@ function useEvents(events, viewProps, stateRef, staticConfig, isHOC, isInsideNat
     if (events.onBlur) {
       viewProps["onBlur"] = events.onBlur;
     }
+    if (Platform.OS === "android") {
+      if (events.onPress) {
+        var _press = events.onPress;
+        viewProps["onClick"] = function (e) {
+          if (!e.nativeEvent || e.nativeEvent.pointerType == null) {
+            _press(e);
+          }
+        };
+      }
+      if (events.onLongPress) {
+        var _longPress = events.onLongPress;
+        viewProps["onLongClick"] = function (e) {
+          if (!e.nativeEvent || e.nativeEvent.pointerType == null) {
+            _longPress(e);
+          }
+        };
+      }
+    }
     if (getIsNativeDesktop()) {
       if (events.onMouseEnter) {
         viewProps["onMouseEnter"] = events.onMouseEnter;
@@ -93,7 +111,7 @@ function useEvents(events, viewProps, stateRef, staticConfig, isHOC, isInsideNat
     return null;
   }
   if (isUsingRNGH) {
-    var callbacksRef = useRef(isUsingRNGH ? {} : null);
+    var callbacksRef = useRef({});
     var gestureRef = useRef(null);
     var useResponderFallback = getIsAndroid() && !(hasRealPressEvents || stateRef.current.hasRealPressEvents) && Boolean(hasPressEvents);
     useMainThreadPressEvents(events, viewProps, useResponderFallback, debugName);
@@ -161,7 +179,8 @@ function useEvents(events, viewProps, stateRef, staticConfig, isHOC, isInsideNat
     }
     return null;
   }
-  useMainThreadPressEvents(events, viewProps, hasPressEvents, debugName);
+  var isPressEnabled = events != null ? hasPressEvents : false;
+  useMainThreadPressEvents(events, viewProps, isPressEnabled, debugName);
   return null;
 }
 function wrapWithGestureDetector(content, gesture, stateRef, isHOC, isCompositeComponent) {
diff --git a/dist/cjs/eventHandling.native.js b/dist/cjs/eventHandling.native.js
index 78ffd237..481bc795 100644
--- a/dist/cjs/eventHandling.native.js
+++ b/dist/cjs/eventHandling.native.js
@@ -77,6 +77,24 @@ function useEvents(events, viewProps, stateRef, staticConfig, isHOC, isInsideNat
     if (events.onBlur) {
       viewProps["onBlur"] = events.onBlur;
     }
+    if (import_react_native.Platform.OS === "android") {
+      if (events.onPress) {
+        var _press = events.onPress;
+        viewProps["onClick"] = function (e) {
+          if (!e.nativeEvent || e.nativeEvent.pointerType == null) {
+            _press(e);
+          }
+        };
+      }
+      if (events.onLongPress) {
+        var _longPress = events.onLongPress;
+        viewProps["onLongClick"] = function (e) {
+          if (!e.nativeEvent || e.nativeEvent.pointerType == null) {
+            _longPress(e);
+          }
+        };
+      }
+    }
     if (getIsNativeDesktop()) {
       if (events.onMouseEnter) {
         viewProps["onMouseEnter"] = events.onMouseEnter;
@@ -135,7 +153,7 @@ function useEvents(events, viewProps, stateRef, staticConfig, isHOC, isInsideNat
     return null;
   }
   if (isUsingRNGH) {
-    var callbacksRef = (0, import_react.useRef)(isUsingRNGH ? {} : null);
+    var callbacksRef = (0, import_react.useRef)({});
     var gestureRef = (0, import_react.useRef)(null);
     var useResponderFallback = getIsAndroid() && !(hasRealPressEvents || stateRef.current.hasRealPressEvents) && Boolean(hasPressEvents);
     (0, import_mainThreadPressEvents.useMainThreadPressEvents)(events, viewProps, useResponderFallback, debugName);
@@ -203,7 +221,8 @@ function useEvents(events, viewProps, stateRef, staticConfig, isHOC, isInsideNat
     }
     return null;
   }
-  (0, import_mainThreadPressEvents.useMainThreadPressEvents)(events, viewProps, hasPressEvents, debugName);
+  var isPressEnabled = events != null ? hasPressEvents : false;
+  (0, import_mainThreadPressEvents.useMainThreadPressEvents)(events, viewProps, isPressEnabled, debugName);
   return null;
 }
 function wrapWithGestureDetector(content, gesture, stateRef, isHOC, isCompositeComponent) {
diff --git a/dist/esm/helpers/matchMedia.native.js b/dist/esm/helpers/matchMedia.native.js
index e4c4d9fc..04612793 100644
--- a/dist/esm/helpers/matchMedia.native.js
+++ b/dist/esm/helpers/matchMedia.native.js
@@ -19,12 +19,11 @@ function matchMediaFallback(query) {
   };
 }
 function setupMatchMedia(_) {
-  if (process.env.NODE_ENV === "development") {
-    if (typeof _ !== "function") {
-      if (!process.env.IS_STATIC) {
-        console.trace(`setupMatchMedia was called without a function, this can cause issues on native`, _);
-      }
+  if (typeof _ !== "function") {
+    if (process.env.NODE_ENV === "development" && !process.env.IS_STATIC) {
+      console.trace(`setupMatchMedia was called without a function, this can cause issues on native`, _);
     }
+    return;
   }
   matchMediaImpl = _;
   globalThis["matchMedia"] = _;
diff --git a/dist/cjs/helpers/matchMedia.native.js b/dist/cjs/helpers/matchMedia.native.js
index 7dcc6de3..85a67889 100644
--- a/dist/cjs/helpers/matchMedia.native.js
+++ b/dist/cjs/helpers/matchMedia.native.js
@@ -49,12 +49,11 @@ function matchMediaFallback(query) {
   };
 }
 function setupMatchMedia(_) {
-  if (process.env.NODE_ENV === "development") {
-    if (typeof _ !== "function") {
-      if (!process.env.IS_STATIC) {
-        console.trace(`setupMatchMedia was called without a function, this can cause issues on native`, _);
-      }
+  if (typeof _ !== "function") {
+    if (process.env.NODE_ENV === "development" && !process.env.IS_STATIC) {
+      console.trace(`setupMatchMedia was called without a function, this can cause issues on native`, _);
     }
+    return;
   }
   matchMediaImpl = _;
   globalThis["matchMedia"] = _;
