diff --git a/build/src/run/ios/appleDevice/client/LockdowndClient.js b/build/src/run/ios/appleDevice/client/LockdowndClient.js
index 4005594..1f049d6 100644
--- a/build/src/run/ios/appleDevice/client/LockdowndClient.js
+++ b/build/src/run/ios/appleDevice/client/LockdowndClient.js
@@ -112,7 +112,7 @@ class LockdowndClient extends _ServiceClient.ServiceClient {
         }
     }
     async startSession(pairRecord) {
-        debug(`startSession: ${pairRecord}`);
+        debug(`startSession: ${JSON.stringify(pairRecord)}`);
         const resp = await this.protocolClient.sendMessage({
             Request: 'StartSession',
             HostID: pairRecord.HostID,
